tests/tensor/test_basic.py

```diff
@@ -2179,6 +2179,51 @@ def test_join_performance(self, ndim, axis, memory_layout, gc, benchmark):
         assert fn(*test_values).shape == (n * 6, n)[:ndim] if axis == 0 else (n, n * 6)
         benchmark(fn, *test_values)
 
+    def test_join_negative_axis_rewrite(self):
+        """Test that constant negative axis is rewritten to positive axis during canonicalization."""
+        v = np.array([[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]], dtype=self.floatX)
+        a = self.shared(v)
+        b = as_tensor_variable(v)
+
+        # Create join with negative axis
+        s = join(-1, a, b)
+
+        # Get the actual Join op node from the graph
+        f = pytensor.function(
+            [], [s], mode=self.mode
+        )  # Use FAST_COMPILE to prevent optimizations from changing the graph structure
+
+        # Find the Join op node in the graph
+        join_nodes = [
+            node for node in f.maker.fgraph.toposort() if isinstance(node.op, Join)
+        ]
+        assert len(join_nodes) == 1, "Expected exactly one Join node in the graph"
+        join_node = join_nodes[0]
```

Comment:
vibe coding is TERRIBLE at writing tests, this is all equivalent to:

```suggestion
        assert s.owner.op.axis==1
```

Since it's not a rewrite level change, but a `make_node` the way you did it. Same below.